src/Workload/Microsoft.Maui.Sdk/Sdk/Microsoft.Maui.Sdk.Before.targets

```diff
@@ -21,4 +21,14 @@
     <_MauiUsingDefaultRuntimeIdentifier>true</_MauiUsingDefaultRuntimeIdentifier>
   </PropertyGroup>
 
+  <!-- Feature switches -->
```

Comment:
I'm not sure if that is the right place to put the feature flag defaults. This feature flag is related to XAML but I think there will be more feature switches that aren't related to it.

But I can definitely imagine moving this to a separate `.targets` file with a target that sets up the feature switches before the `RunILLink` target runs.

Reply:
@simonrozsival do you know offhand if any of the trimmer warnings so far are from `Microsoft.Maui.dll` / `Core.csproj`, or are they all from `Microsoft.Maui.Controls.dll` & `Microsoft.Maui.Controls.Xaml.dll`?

It might make sense to put the trimmer switches in "Controls", as it is possible to create an app that uses parts of MAUI and not `Microsoft.Maui.Controls.dll`.

Reply:
The warnings we're getting are mostly from `Microsoft.Maui.Controls.dll`, but for example the DI related ones are in `Microsoft.Maui.dll` (for example the MauiFactory related code). Yesterday we talked about putting the existing reflection-based image source service resolution code behind a feature flag. Wouldn't we need a new separate setup with a separate ILLink.Subsitutions.xml to introduce that feature flag if we moved this feature flag from Core into Controls?

Reply:
I think another reason to move from `src/Workload/Microsoft.Maui.Sdk/Sdk/Microsoft.Maui.Sdk.Before.targets`, is this file is in the workload, and wouldn't be able to be updated independently with `$(MauiVersion)` or the `PackageReference` in the project template.

@mattleibow is there a `.targets` file that would also include a way for feature flags to exist in Core/`Microsoft.Maui.dll`?

Reply:
I see. I can move the code to `Microsoft.Maui.Controls.targets`. Even with the code in that targets file, we can still define IL substitutions for the `RuntimeFeature` class which will stay in Core.

Reply:
I think that's the way to go for now, I was looking in:

"C:\Program Files\dotnet\library-packs\Microsoft.Maui.Controls.Core.8.0.3.nupkg"

And there are no `.targets` files in there currently.